### ISSUE TITLE
Broken plugin with IDEA 2016.2

### ISSUE THREAD

**Issue by eekboom** [uwolfer/gerrit-intellij-plugin#210]
The gerrit plugin does not compile with the latest version of IDEA's git4idea plugin (master on https://github.com/JetBrains/intellij-community/tree/master/plugins/git4idea) which will be part of IDEA 2016.2.

Classes used by the gerrit plugin have been renamed and (constructor) parameters have changed.

**Comment by uwolfer** [comment]
Where do you find 2016.2 EAP download links? I cannot find it on the usual page: https://confluence.jetbrains.com/display/IDEADEV/IDEA+2016.2+EAP

**Comment by eekboom** [comment]
It has not been published yet. (Dmitry Jemerov granted me access to JetBrains' internal TeamCity installation many years ago and I am mostly working with nightly builds of IDEA 
ever since.)
That's why I asked in my original mail if you maybe have access to those builds.

But you can already find the updated 2016.2 sources of the git4idea plugin in the master branch of its github repo.
I also just asked JetBrains if they could grant you access to nightly builds, too.

So, all in all, no hurry. I just thought I tip you off and you could maybe already look at the updated git4idea.

Regards
Stephen

Am 12.04.2016 um 20:12 schrieb Urs Wolfer:

**Comment by uwolfer** [comment]
> It has not been published yet. (Dmitry Jemerov granted me access to JetBrains' internal TeamCity installation many years ago and I am mostly working with nightly builds of IDEA
ever since.)
> That's why I asked in my original mail if you maybe have access to those builds.

Ah, now I understand why you asked this question :)

> But you can already find the updated 2016.2 sources of the git4idea plugin in the master branch of its github repo.

Yes, but compiling whole IntelliJ from source takes quite some time.

> I also just asked JetBrains if they could grant you access to nightly builds, too.

Great, thanks.

> So, all in all, no hurry. I just thought I tip you off and you could maybe already look at the updated git4idea.

Yes, but from my experience once an EAP is out, a lot of people are updating and when the plugin is broken, I get a lot of "automated" exception reports. So it's better to fix it as early as possible. Thanks a lot for your notification in this case.

**Comment by lexbel** [comment]
I faced with this issue that on 2016.2 EAP plugin is not working. 
I put stacktrace here because I couldn't submit it due to corporate proxy:

```
com.urswolfer.intellij.plugin.gerrit.ui.RepositoryChangesBrowserProvider$GerritRepositoryChangesBrowser.getDiffAction()Lcom/intellij/openapi/vcs/changes/actions/diff/ShowDiffAction;
java.lang.NoSuchMethodError: com.urswolfer.intellij.plugin.gerrit.ui.RepositoryChangesBrowserProvider$GerritRepositoryChangesBrowser.getDiffAction()Lcom/intellij/openapi/vcs/changes/actions/diff/ShowDiffAction;
    at com.urswolfer.intellij.plugin.gerrit.ui.RepositoryChangesBrowserProvider.get(RepositoryChangesBrowserProvider.java:87)
    at com.urswolfer.intellij.plugin.gerrit.ui.GerritToolWindow.createToolWindowContent(GerritToolWindow.java:71)
    at com.urswolfer.intellij.plugin.gerrit.ui.GerritToolWindowFactory.createToolWindowContent(GerritToolWindowFactory.java:36)
    at com.intellij.openapi.wm.impl.ToolWindowImpl.ensureContentInitialized(ToolWindowImpl.java:586)
    at com.intellij.openapi.wm.impl.ToolWindowManagerImpl$12.run(ToolWindowManagerImpl.java:561)
    at com.intellij.openapi.application.TransactionGuardImpl$7.run(TransactionGuardImpl.java:294)
    at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:326)
    at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:310)
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
    at java.awt.EventQueue.access$500(EventQueue.java:97)
    at java.awt.EventQueue$3.run(EventQueue.java:709)
    at java.awt.EventQueue$3.run(EventQueue.java:703)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
    at com.intellij.ide.IdeEventQueue.a(IdeEventQueue.java:828)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:663)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:393)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

```

I fixed this issue just by upgrading Intelliij Idea version in `gradle.properties` to the last one.
`ideaVersion=IC-162.74.16`

**Comment by uwolfer** [comment]
@lexbel: Thanks for your hint. So when rebuilding the plugin, everything works fine?

**Comment by lexbel** [comment]
@uwolfer yes, at least I haven't found any other issues.

**Comment by uwolfer** [comment]
I have released a new build for IntelliJ 2016.2 EAP. Please refresh plugin list in settings and update the plugin manually if your IDE does not show an update notice.

**Comment by uwolfer** [comment]
If you also get this error, please try to following and report back if helps or not (I have received this hint by a JetBrains developer):

> "Check for update" action forces check for updates for sure and it is accessible through "Find action" or Settings->Appeaerance&Behavior->System Settings->Updates.

If it does not help, I can give you a contact at JetBrains in order to analyze this issue (and thus help other users in future). Thanks.
